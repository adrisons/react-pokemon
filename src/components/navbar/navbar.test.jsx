import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import logo from "../../assets/pokemon-logo.png";
import Navbar from "./navbar";

describe("WHEN: Navbar", () => {
  const component = renderer.create(
    <Router>
      <Navbar />
    </Router>
  );
  const tree = component.toJSON();
  it("THEN: should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  it("THEN: img must have src = pokemon-logo.png and alt = logo", () => {
    render(
      <Router>
        <Navbar />
      </Router>
    );
    const img = screen.getByRole("img");
    expect(img).toHaveAttribute("src", logo);
    expect(img).toHaveAttribute("alt", "logo");
  });
});
