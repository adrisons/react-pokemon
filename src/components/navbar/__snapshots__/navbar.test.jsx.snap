// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN: Navbar THEN: should match snapshot 1`] = `
<div
  className="navbar"
>
  <a
    className="logo-container"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      className="logo"
      src="pokemon-logo.png"
    />
  </a>
</div>
`;
